### ISSUE TITLE
2.13.0 makes my app crash using withThrowingTaskGroup

### ISSUE THREAD

**Issue by florianmari** [supabase/supabase-swift#413]
# Bug report

Hello,

Seems I have a crash when I have updated SDK to the last version, probably because of the addition of Task helpers or maybe I'm doing something wrong but it seems I cannot use `withThrowingTaskGroup` like this example

```
let images = try await withThrowingTaskGroup(of: Story.self, returning: [Story].self) { taskGroup in
            let stories = try await client.stories(userId: profileId)
            for story in stories {
                if client.checkIfValidStory(date: story.date) {
                    taskGroup.addTask { try await client.story(userId: profileId, info: story)! }
                }
            }
            
            var images = [Story]()
            for try await result in taskGroup {
                images.append(result)
            }
            return images.sortedByDate()
        }
```

This basically calls 2 func that uses storage api, one to list all files and second to download file

<img width="1495" alt="image" src="https://github.com/supabase/supabase-swift/assets/1582796/04671fa2-4806-4883-96e9-569e6e1b3996">

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Cannot use `withThrowingTaskGroup`

## To Reproduce

Use `withThrowingTaskGroup`

## Expected behavior

Should work as before concurrently

## System information

//

**Comment by grdsdev** [comment]
Hi @florianmari 

Thanks for bringing up this, what is happening is that we're now using task locals for the internal logging of the library.

Doing some research I found a few issues on Swift compiler referencing this:
- https://github.com/apple/swift/issues/73217

We should not use task locals within the `withThrowingTaskGroup`'s `body`, that happens at the line `let stories = try await client.stories(userId: profileId)`.

In your code I think if you move that above line out of the `withThrowingTaskGroup`'s body it will work.

```swift
let stories = try await client.stories(userId: profileId) // this line was moved

let images = try await withThrowingTaskGroup(of: Story.self, returning: [Story].self) { taskGroup in
  for story in stories {
    if client.checkIfValidStory(date: story.date) {
      taskGroup.addTask { try await client.story(userId: profileId, info: story)! }
    }
  }
  
  var images = [Story]()
  for try await result in taskGroup {
    images.append(result)
  }
  return images.sortedByDate()
}
```

---

This Swift bug has been fixed in https://github.com/apple/swift/pull/73978 but unfortunately that will land only on **Swift 6.0**.

If the issue persists after you change your code to the suggestion above, I'll revert that task locals change from the logging layer.

Thank you!
